### PR TITLE
feat(chips): add ripple to chips

### PR DESCRIPTION
**Pull request by tinayuangao**

### CODE REVIEW

File: src/lib/chips/chip.ts [hunk 1]
```diff
@@ -93,10 +110,26 @@ export class MatChipTrailingIcon {}
     '(focus)': '_hasFocus = true',
     '(blur)': '_blur()',
   },
-
 })
 export class MatChip extends _MatChipMixinBase implements FocusableOption, OnDestroy, CanColor,
-    CanDisable {
+  CanDisable, CanDisableRipple, RippleTarget {
```
Comment (mmalerba) [inline src/lib/chips/chip.ts]: nit: continuation, indent 4

File: src/lib/chips/chip.ts [hunk 2]
```diff
@@ -161,7 +194,7 @@ export class MatChip extends _MatChipMixinBase implements FocusableOption, OnDes
 
   /** Emitted when the chip is selected or deselected. */
   @Output() readonly selectionChange: EventEmitter<MatChipSelectionChange> =
-      new EventEmitter<MatChipSelectionChange>();
+    new EventEmitter<MatChipSelectionChange>();
```
Comment (mmalerba) [inline src/lib/chips/chip.ts]: nit: continuation, indent 4

File: src/lib/chips/chip.ts [hunk 3]
```diff
@@ -188,10 +221,23 @@ export class MatChip extends _MatChipMixinBase implements FocusableOption, OnDes
...
+
+    this._chipRipple = new RippleRenderer(this, ngZone, _elementRef, platform);
+    this._chipRipple.setupTriggerEvents(_elementRef.nativeElement);
+
+    if (globalOptions) {
+      this.rippleConfig = {
+        speedFactor: globalOptions.baseSpeedFactor,
+        animation: globalOptions.animation,
+      };
+    }
```
Comment (DevVersion) [inline src/lib/chips/chip.ts]: Thinking about that again, it's weird that we need to repeat that so often, and even need to keep track of the global options. Just wanted to add that here, nothing to fix in this PR actually.

Comment (DevVersion) [reply src/lib/chips/chip.ts]: @tinayuangao Just saw, that this does not include `terminateOnPointerUp` yet. This would be important to add here as well.

**Pull request merged**